connectors/memory/memory.go

```diff
@@ -326,13 +334,51 @@ func (c *Connector) Remove(_ context.Context, ei *dosa.EntityInfo, values map[st
 	return nil
 }
 
+// RemoveRange removes all of the elements in the range specified by the entity info and the column conditions.
+func (c *Connector) RemoveRange(_ context.Context, ei *dosa.EntityInfo, columnConditions map[string][]*dosa.Condition) error {
+	c.lock.Lock()
+	defer c.lock.Unlock()
+
+	partitionRange := c.findRange(ei, columnConditions)
+	if partitionRange == nil {
+		return nil
+	}
+
+	partitionRef := partitionRange.entityRef[partitionRange.partitionKey]
```

Comment:
Did you look at making methods on partitionRange that do the heavy lifting? Something like:
```
func (p *partitionRange) deleteRange(er *[]map[string]dosa.FieldValue) {
    er = append(er[:p.start], er[p.end:]...)
```
The caller would then do:
```
partitionRange.deleteRange(&entityRef[partitionKey])
```